backend/src/v4/models/ref.js

```diff
@@ -33,26 +33,17 @@ Ref.getRefNodes = async (account, model, branch, revision, projection) => {
 };
 
 Ref.getSubModels = async (account, model, branch, revision, callbackProm) => {
-	const refs = await Ref.getRefNodes(account, model, branch, revision, {owner: 1, project: 1, _rid: 1});
+	const settings = await ModelSetting.findModelSettingById(account, model);
 	const subModelArr = [];
-	for(let i = 0; i < refs.length; ++i) {
-		const {owner, project, _rid} = refs[i];
-		let refBranch, refRev;
-		if (utils.uuidToString(_rid) === C.MASTER_BRANCH) {
-			refBranch = C.MASTER_BRANCH_NAME;
-		} else {
-			refRev = utils.uuidToString(_rid);
-		}
+	for(let i = 0; i < settings.subModels.length; ++i) {
```

Comment:
Are we confident about subModels would always exist? (e.g. on situations where federation is created bu no entry, or the model is not found, or the model is actually a container?)

Wondering if it would be better if we do a chgeck before we do the looping.